rpcs/blockService.go

```diff
@@ -382,7 +405,14 @@ func (bs *BlockService) rawBlockBytes(round basics.Round) ([]byte, error) {
 		}
 	default:
 	}
-	return RawBlockBytes(bs.ledger, round)
+	if bs.memoryUsed > bs.memoryCap {
+		return nil, errMemoryAtCapacity{used: bs.memoryUsed, capacity: bs.memoryCap}
+	}
+	data, err := RawBlockBytes(bs.ledger, round)
+	if err == nil {
```

Comment:
I realize that data is not nil when err is nil, but it's the data that is allocated or not.
```suggestion
	if data != nil {
```

Reply:
sorry I think I hit my head this morning.